include/roboteam_ai/stp/new_plays/Pass.h

```diff
@@ -37,22 +37,27 @@ class Pass : public Play {
      */
     void calculateInfoForRoles() noexcept override;
 
-    /**
-     * Calculates n defensive positions for the roles to defend
-     * @param numberOfDefenders
-     * @param world
-     * @param enemyRobots
-     * @return A vector of defend positions
-     */
-    std::vector<Vector2> calculateDefensivePositions(int numberOfDefenders, world_new::World* world, std::vector<world_new::view::RobotView> enemyRobots);
-
     /**
      * Gets the play name
      */
     const char* getName() override;
 
    protected:
     bool shouldRoleSkipEndTactic() override;
+
+    /**
+     * Decide whether or not we need to switch to a different pass location. Currently based on:
+     *      The ratio of the scores
+     *      The proximity of the locations
+     *      The proximity of the robot to either location
+     * @param candidatePosition the position suggested by the current iteration of the pass algorithm
+     * @param currentPosition the position suggested by the previous iteration of the pass algorithm
+     * @return true if desirable to switch
+     */
+    std::pair<Vector2, double> compareNewLocationToCurrentLocation(Vector2 currentPosition, Vector2 candidatePosition);
+
+    Vector2 currentPassLocation;
+    double currentPassScore;
```

Comment:
Are you sure you initialize these before using them? if not you should default initialize

Reply:
Yes, in the constructor I do!